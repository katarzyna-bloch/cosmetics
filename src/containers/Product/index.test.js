import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Product } from '.'

it('should render correctly', () => {
  const props = {
    fetchProducts: jest.fn(),
    product: {
      brand: 'Dove',
      name: 'Test',
      img: 'img.jpg',
      volume: '20ml',
      price: '10z≈Ç',
      description: 'Opis produktu',
      },
    match: {
      params: {
        id: '1',
      },
    },
  }
  const wrapped = mount(<Product {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  expect(props.fetchProducts).toHaveBeenCalled()
})
