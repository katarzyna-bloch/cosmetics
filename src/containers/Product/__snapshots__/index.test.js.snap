// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Product
  fetchProducts={
    [MockFunction] {
      "calls": Array [
        Array [
          "1",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  loading={false}
  match={
    Object {
      "params": Object {
        "id": "1",
      },
    }
  }
  product={
    Object {
      "brand": "mokosh",
      "category": Object {
        "heading": "Profesjonalne SPA w domowym zaciszu",
        "id": "1",
        "name": "Pielęgnacja",
        "type": "care",
        "url": "/categories/care",
      },
      "categoryId": "1",
      "description": "Opis produktu",
      "id": "7",
      "img": "https://www.topestetic.pl/img/",
      "name": "Bambusowa Szczotka do Masażu na Sucho",
      "price": "19 zł",
      "volume": "1 szt.",
    }
  }
>
  <styled.div>
    <div
      className="sc-bdvvaa hugagk"
    >
      <Breadcrumb
        values={
          Array [
            Object {
              "label": "Pielęgnacja",
              "link": "/categories/care",
            },
            Object {
              "label": "Bambusowa Szczotka do Masażu na Sucho",
            },
          ]
        }
      />
      <Product
        brand="mokosh"
        category={
          Object {
            "heading": "Profesjonalne SPA w domowym zaciszu",
            "id": "1",
            "name": "Pielęgnacja",
            "type": "care",
            "url": "/categories/care",
          }
        }
        categoryId="1"
        description="Opis produktu"
        id="7"
        img="https://www.topestetic.pl/img/"
        name="Bambusowa Szczotka do Masażu na Sucho"
        price="19 zł"
        volume="1 szt."
      />
    </div>
  </styled.div>
</Product>
`;
